app/src/controllers/object.js

```diff
@@ -617,16 +617,18 @@ const controller = {
    */
   async searchObjects(req, res, next) {
     // TODO: Handle no database scenarios via S3 ListObjectsCommand?
-    // TODO: Consider metadata/tagging query parameter design here?
+    // TODO: Consider tagging query parameter design here
     // TODO: Consider support for filtering by set of permissions?
     // TODO: handle additional parameters. Eg: deleteMarker, latest
     try {
       const objIds = mixedQueryToArray(req.query.objId);
+      const metadata = getMetadata(req.headers);
       const params = {
         id: objIds ? objIds.map(id => addDashesToUuid(id)) : objIds,
         name: req.query.name,
         path: req.query.path,
         mimeType: req.query.mimeType,
+        metadata: metadata && Object.keys(metadata).length ? metadata : undefined,
```

Comment:
this && stuff was in case they provide a key with no value, right?
by the way, i wonder if we need to urldecode keys and values.

Reply:
At the controller level, we need to check for two things:

1. 1) Is metadata truthy (aka does it exist)
2. 2) Is the object actually populated with "something"

We need the second check because in the event there is nothing/undefined, we want to outright skip that portion of logic in the later query building phase.

Url encoding and decoding for headers should be fine for metadata; I didn't see too many examples in the wild where we had to sanitize metadata headers.